### ISSUE TITLE
When SLF4J and Logback are initialized on multiple threads in parallel, startup may fail due to SubstituteLoggerFactory being considered to be a competing LoggerFactory implementation

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-boot#37484]
Forward port of issue #37477 to 3.1.x.

**Comment by drnesic** [comment]
We tested upgrade from spring-boot 3.1.4 to 3.2.1 and this problem surfaced again, will the solution be forward ported to 3.2.x? Alternatively, is there a workaround?

**Comment by wilkinsona** [comment]
[The fix is in 3.2.x](https://github.com/spring-projects/spring-boot/blob/8bb60eef026fc8787b5ac763bcb9c532f3fcb5f9/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/logback/LogbackLoggingSystem.java#L409-L418). If you're seeing the problem with 3.2.1, please open a new issue with a minimal example that reproduces it.
